onert-micro/onert-micro/include/onert-micro-train.h

```diff
@@ -0,0 +1,235 @@
+ *   1-1. (optional) configure training options
+ * 2. feed training input / output(e.g. label) data (cf. unit of a step)
+ * 3. train a step
+ * 4. check loss
+ *   4-0. save checkpoint for recovery/resume training
+ *   4-1. no more traning -> go to 5
+ *   4-2. more training -> go to 2
+ * 5. save current state to inference model
+ * 6. inference with inference model
+
+// sample example
+// 0. create context
+om_context *ctx;
+om_create_context(&ctx);
+
+// 1. load model
+om_load_model_from_file(ctx, MODEL_PATH);
+// 1-1. (optional, TBD) configure training options
+
+om_train_compile(ctx);
+
+float training_input[BATCH_SIZE*INPUT_SIZE];
+float training_label[BATCH_SIZE*OUTPUT_SIZE];
+
+// main training loop
+for(int epoch=0; epoch < NUM_EPOCHS; epoch++) {
+  for(int step=0; step < NUM_BATCHES ; step++) {
+    // prepare this steps's intput/label
+    memcpy(training_input, train_input_data + THIS_BATCH_OFFSET, BATCH_SIZE*INPUT_SIZE);
+    memcpy(training_output, train_output_data + THIS_BATCH_OFFSET, BATCH_SIZE*OUTPUT_SIZE);
+
+    // 2. feed training input / output
+    om_train_set_input(ctx, 0 , training_input, BATCH_SIZE*INPUT_SIZE);
+    om_train_set_expected(ctx, 0 , training_input,BATCH_SIZE*INPUT_SIZE);
+    // 3. train a step
+    om_train_single_step(ctx);
+  }
+  // 4. check loss
+  float loss;
+  om_train_get_loss(ctx, 0, &loss);
+
+  if(loss > TARGET_LOSS) {
+    om_train_save_as_checkpoint(ctx, PATH);
+  }
+  else {
+    om_train_save_as_inferencemodel(ctx, PATH);
+  }
+}
+*/
+
+
+
+
+/**
+ * @brief Result values returned from a call to an API function
+ */
+typedef enum
+{
+  /** Successful */
+  OM_STATUS_NO_ERROR = 0,
+  /**
+   * An error code for general use.
+   * Mostly used when there is no specific value for that certain situation.
+   */
+  OM_STATUS_ERROR = 1,
+} OM_STATUS;
+
+struct om_context;
+
+/**
+ * @brief Create a new context.
+ *
+ * @param[out]  cxt The context to be created
+ * @return      OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_create_context(om_context **cxt);
+
+/**
+ * @brief Load model from model dir
+ *
+ * @param[in] ctx  context loading the given model dir
+ * @param[in] path Path to the model or to the dir which contains training artifacts
+ *
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_load_model_from_file(om_context *ctx, const char *path);
+
+/**
+ * @brief Load checkpoint from checkpoint dir
+ *
+ * @param[in] ctx  context loading the given checkpoint
+ * @param[in] path Path to the checkpoint dir which contains all the checkpoint files
+ *
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_load_checkpoint_from_file(om_context *ctx, const char *path);
+
+
+/**
+ * @brief Prepare context to be ready for training
+ * @note  The context will be entered into training mode
+ *
+ * @param[in] ctx The context to be prepared for training
+ *
+ * @return  @c OM_STATUS_NO_ERROR if successful
+ */
+OM_STATUS om_train_compile(om_context *ctx);
```

Comment:
some more parameter need to be expanded, as I introduced API set in last meeting.(late, loss, metrix...)

Reply:
I will add config api like `set_train_info`. But note that this config is optional and basically, circle model itself contains training info(loss,...)